cpp/tests/copying/scatter_tests.cu

```diff
@@ -609,3 +611,285 @@ TEST_F(ScatterStringsTests, ScatterScalarNoNulls)
+    std::vector<std::unique_ptr<cudf::scalar>> scalar_vect;
+    scalar_vect.push_back(std::move(scalar));
+    cudf::test::fixed_width_column_wrapper<int64_t> target(                     {   2,     2,     3,     4,   11,   12,     7,    7,   10,    10});
+    cudf::test::fixed_width_column_wrapper<cudf::experimental::bool8> mask({true,  false, false, false, true, true, false, true, true, false});
+    auto target_table = cudf::table_view({target});
+
+    EXPECT_THROW(cudf::experimental::boolean_mask_scatter(scalar_vect, target_table, mask), cudf::logic_error);
+}
+
+TEST_F(BooleanMaskScatterScalarFails, BooleanMaskTypeMismatch)
+{
+    auto scalar = cudf::make_numeric_scalar(cudf::data_type(cudf::data_type{cudf::experimental::type_to_id<int32_t>()}));
+    std::vector<std::unique_ptr<cudf::scalar>> scalar_vect;
+    scalar_vect.push_back(std::move(scalar));
+    cudf::test::fixed_width_column_wrapper<int32_t> target(                     {   2,     2,     3,     4,   11,   12,     7,    7,   10,    10});
+    cudf::test::fixed_width_column_wrapper<int8_t> mask({true,  false, false, false, true, true, false, true, true, false});
+    auto target_table = cudf::table_view({target});
+
+    EXPECT_THROW(cudf::experimental::boolean_mask_scatter(scalar_vect, target_table, mask), cudf::logic_error);
+}
+
+TEST_F(BooleanMaskScatterScalarFails, BooleanMaskTargetSizeMismatch)
+{
+    auto scalar = cudf::make_numeric_scalar(cudf::data_type(cudf::data_type{cudf::experimental::type_to_id<int32_t>()}));
+    std::vector<std::unique_ptr<cudf::scalar>> scalar_vect;
+    scalar_vect.push_back(std::move(scalar));
+    cudf::test::fixed_width_column_wrapper<int32_t> target(                     {   2,     2,     3,     4,   11,   12,     7,    7,   10,    10});
+    cudf::test::fixed_width_column_wrapper<cudf::experimental::bool8> mask({true,  false, false, false, true, true, false, true, true});
+    auto target_table = cudf::table_view({target});
+
+    EXPECT_THROW(cudf::experimental::boolean_mask_scatter(scalar_vect, target_table, mask), cudf::logic_error);
+}
+
+TEST_F(BooleanMaskScatterScalarFails, NumberOfColumnAndScalarMismatch)
+{
+    auto scalar = cudf::make_numeric_scalar(cudf::data_type(cudf::data_type{cudf::experimental::type_to_id<int32_t>()}));
+    std::vector<std::unique_ptr<cudf::scalar>> scalar_vect;
+    scalar_vect.push_back(std::move(scalar));
+    scalar_vect.push_back(std::move(scalar));
+    cudf::test::fixed_width_column_wrapper<int32_t> target(                     {   2,     2,     3,     4,   11,   12,     7,    7,   10,    10});
+    cudf::test::fixed_width_column_wrapper<cudf::experimental::bool8> mask({true,  false, false, false, true, true, false, true, true});
+    auto target_table = cudf::table_view({target});
+
+    EXPECT_THROW(cudf::experimental::boolean_mask_scatter(scalar_vect, target_table, mask), cudf::logic_error);
+}
+
```

Comment:
Is there a test where the number of true values is greater than the input size?

Reply:
Ahh, missed that. I have added it now.